bootstrap/handlers/secret/client/vaultclient.go

```diff
@@ -0,0 +1,90 @@
+//
+// Copyright (c) 2019 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License. You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software distributed under the License
+// is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+// or implied. See the License for the specific language governing permissions and limitations under
+// the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+package client
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/edgexfoundry/go-mod-bootstrap/config"
+
+	"github.com/edgexfoundry/go-mod-core-contracts/clients/logger"
+
+	"github.com/edgexfoundry/go-mod-secrets/pkg"
+	"github.com/edgexfoundry/go-mod-secrets/pkg/providers/vault"
+	"github.com/edgexfoundry/go-mod-secrets/pkg/token/authtokenloader"
+	"github.com/edgexfoundry/go-mod-secrets/pkg/token/fileioperformer"
+)
+
+// Vault is the structure to get the secret client from go-mod-secrets vault package
+type Vault struct {
+	ctx    context.Context
+	config *vault.SecretConfig
+	lc     logger.LoggingClient
+}
+
+// NewVault is the constructor for Vault in order to get the Vault secret client
+func NewVault(
+	ctx context.Context,
+	config vault.SecretConfig,
+	lc logger.LoggingClient) Vault {
+	return Vault{
+		ctx:    ctx,
+		config: &config,
```

Comment:
Why store as a reference?  You're not updating it anywhere and it's being passed into NewVault() as a value.

Reply:
will change it as a value